common/src/main/java/com/wynntils/handlers/actionbar/ActionBarHandler.java

```diff
@@ -70,18 +72,30 @@ public void onActionBarUpdate(ChatPacketReceivedEvent.GameInfo event) {
         Arrays.stream(ActionBarPosition.values()).forEach(pos -> processPosition(pos, positionMatches));
 
         StyledText newContentBuilder = StyledText.EMPTY;
+        // vanilla segments have three spaces between each segment, regardless of content
         if (!lastSegments.get(ActionBarPosition.LEFT).isHidden()) {
             newContentBuilder = newContentBuilder.append(positionMatches.get(ActionBarPosition.LEFT));
         }
         if (!lastSegments.get(ActionBarPosition.CENTER).isHidden()) {
+            newContentBuilder = newContentBuilder.append(STANDARD_PADDING);
             newContentBuilder = newContentBuilder.append(positionMatches.get(ActionBarPosition.CENTER));
+            newContentBuilder = newContentBuilder.append(STANDARD_PADDING);
         } else {
             // Add padding
             newContentBuilder = newContentBuilder.append(CENTER_PADDING);
         }
         if (!lastSegments.get(ActionBarPosition.RIGHT).isHidden()) {
-            newContentBuilder = newContentBuilder.append(positionMatches.get(ActionBarPosition.RIGHT));
+            StyledText right = positionMatches.get(ActionBarPosition.RIGHT);
+
+            // Rare case where the user has 100% curse charge, which makes StyledText remove the color code
```

Comment:
This seems like the wrong solution for two reasons.
1) It forces right to be aqua, assuming that it is always mana. I'm not sure it always is..? Can't it show different values e.g. when swimming? And even if it is, hard-coding it like this feels wrong.

2) We should have StyledText properly split up the string to include all formatting codes needed for each part. Iirc, we now have such a split function, but we did not at the time when the ActionBarHandler was written. So we should use that instead of this workaround.

And if we do not have such a function, we need to write one. It is the responsibility of StyledText to be able to include all relevant formatting codes.